### DISCUSSION TITLE
Combination of remote.mount and remote.uncache?

### DISCUSSION THREAD

**Discussion by ms-vaas-digital** [seaweedfs/seaweedfs#5715] [Q&A]
Is it possible to mount a remote storage (remote.mount) while at the same time specifying that it should be initially uncached (remote.uncached)?

I'm testing SeaweedFS mounted to Azure Blob Storage. It can take quite some time to fire up with Docker if there are already millions of files in the remote storage that need to be cached locally before the drive is mounted and ready.

**Comment by chrislusf** [comment]
yes

**Reply by ms-vaas-digital** [reply]
My understanding from the documentation (https://github.com/seaweedfs/seaweedfs/wiki/Cache-Remote-Storage) is that I must first use remote.mount to mount the remote storage. But there seems to be no argument that I can send with the command to mount the remote storage in an initially uncached state?

I've tried to immediately afterwards execute the remote.uncache command just after remote.mount but this won't succeed untill the remote storage has been fully mounted, ie. millions of files have been downloaded and cached locally.

So how should i combine remote.mount and remote.uncache to mount a remote storage as initially uncached?

**Reply by chrislusf** [reply]
`remote.mount` only pulls metadata. File content is not included.

**Reply by ms-vaas-digital** [reply]
We're using the relatively new support for Azure Blob Storage as specified by the remote.configure command.

I'm using this exact script to remote mount the Azure Blob Storage account:

```
#!/bin/sh

set -e

echo "Mounting Azure Storage"
echo "remote.configure -name=azure -type=azure -azure.account_name=filestorageaccount -azure.account_key=XYZ -s3.storage_class=STANDARD -s3.support_tagging=false" | /usr/bin/weed shell
echo "remote.mount -dir=/A/B -remote=azure/bucket/A/B -nonempty" | /usr/bin/weed shell
exit 0
```

What I've seen in my testing today is that the last remote.mount command doesn't complete untill after millions of files have been downloaded and cached locally.

Could this be an implementation issue specifically when mounting Azure Blob Storage?

**Reply by chrislusf** [reply]
How to tell millions of files are downloaded?
